flyingsnake/__main__.py

```diff
@@ -98,98 +123,112 @@ def flyingsnake(input_file: str,
 
     c.echo("Parsing world...")
     world = li.World.create_from_file(input_file)
-
+    if (min_x != DEFAULT_COORDS or min_y != DEFAULT_COORDS
+        or region_height != DEFAULT_REGION_VALUE or region_width != DEFAULT_REGION_VALUE):
+        min_x, min_y, max_x, max_y = get_region_size(world=world, min_x=min_x, min_y=min_y,
+                                                     region_width=region_width, region_height=region_height)
+        c.echo(f"Only rendering world coordinates ({min_x}, {min_y}) to ({max_x}, {max_y}")
+    else:
+        min_x, min_y, max_x, max_y = 0, 0, world.size.x, world.size.y
+    width = max_x - min_x
+    height = max_y - min_y
     if draw_background:
         c.echo("Drawing the background...")
-        background = Image.new("RGBA", (world.size.x, world.size.y))
+        background = Image.new("RGBA", (width, height))
         draw = ImageDraw.Draw(background)
-        draw.rectangle(((0, 0), (world.size.x, world.underground_level)), tuple(colors["Globals"]["Sky"]))
-        draw.rectangle(((0, world.underground_level + 1), (world.size.x, world.cavern_level)),
+        draw.rectangle(((0, 0), (width, world.underground_level)), tuple(colors["Globals"]["Sky"]))
+        draw.rectangle(((0, world.underground_level + 1), (width, world.cavern_level)),
                        tuple(colors["Globals"]["Earth"]))
-        draw.rectangle(((0, world.cavern_level + 1), (world.size.x, world.size.y - 192)),
+        draw.rectangle(((0, world.cavern_level + 1), (width, height - 192)),
                        tuple(colors["Globals"]["Rock"]))
-        draw.rectangle(((0, world.size.y - 191), (world.size.x, world.size.y)), tuple(colors["Globals"]["Hell"]))
+        draw.rectangle(((0, height - 191), (width, height)), tuple(colors["Globals"]["Hell"]))
```

Comment:
If you try to render a small world region, you'll see that both the "hell" background and the "overworld" background will be present in the render, even if the parts containing them aren't rendered:

![](https://user-images.githubusercontent.com/1540885/65966966-1f52ee00-e461-11e9-8e6f-63ce57c6f8e8.png)

There's something that needs to be changed in the background rendering phase...